app/controllers/concerns/foreman/controller/puppet/hosts_controller_extensions.rb

```diff
@@ -0,0 +1,164 @@
+module Foreman::Controller::Puppet::HostsControllerExtensions
+  extend ActiveSupport::Concern
```

Comment:
What's the benefit of extracting it to mixing first as opposed to creating a new controller? What prevents us introducing the new controller already? If there are some parts that can't be done in the controller right now, it could still be better to keep just those in concern but the rest could already be in a new controller.